dhcpv6/ztp/parse_remote_id.go

```diff
@@ -0,0 +1,80 @@
+package ztpv6
+
+import (
+	"fmt"
+	"regexp"
+
+	"github.com/insomniacslk/dhcp/dhcpv6"
+)
+
+var (
+	// Arista Port, Vlan Pattern
+	aristaPVPattern = regexp.MustCompile("Ethernet(?P<port>[0-9]+):(?P<vlan>[0-9]+)")
+	// Arista Slot, Mod, Port Pattern
+	aristaSMPPattern = regexp.MustCompile("Ethernet(?P<slot>[0-9]+)/(?P<module>[0-9]+)/(?P<port>[0-9]+)")
+)
+
+// CircuitID represents the structure of network vendor interface formats
+type CircuitID struct {
+	Slot    string
+	Module  string
+	Port    string
+	SubPort string
+	Vlan    string
+}
+
+// ZTPRemoteId will parse the RemoteId Option data for Vendor Specific data
+func ZTPRemoteId(packet dhcpv6.DHCPv6) (CircuitID, error) {
+	// Need to decapsulate the packet after multiple relays in order to reach RemoteId data
+	inner, err := dhcpv6.DecapsulateRelayIndex(packet, -1)
+	if err != nil {
+		return CircuitID{}, fmt.Errorf("failed to decapsulate relay index: %v", err)
+	}
+	var remoteID string
+	var cid CircuitID
+	rid := inner.GetOneOption(dhcpv6.OptionRemoteID)
+	if rid != nil {
+		remoteID = string(rid.(*dhcpv6.OptRemoteId).RemoteID())
+		cid, err = parseZtpRemoteID(remoteID)
+		if err != nil {
+			return CircuitID{}, err
+		}
+	}
+	return cid, nil
+}
+
+func parseZtpRemoteID(remoteID string) (CircuitID, error) {
+	var names []string
+	var matches []string
+
+	switch {
+	case aristaPVPattern.MatchString(remoteID):
+		matches = aristaPVPattern.FindStringSubmatch(remoteID)
+		names = aristaPVPattern.SubexpNames()
+	case aristaSMPPattern.MatchString(remoteID):
+		matches = aristaSMPPattern.FindStringSubmatch(remoteID)
+		names = aristaSMPPattern.SubexpNames()
+	}
+
+	circuit := CircuitID{}
+	for i, match := range matches {
+		switch names[i] {
+		case "port":
+			circuit.Port = match
+		case "slot":
+			circuit.Slot = match
+		case "module":
+			circuit.Module = match
+		case "subport":
+			circuit.SubPort = match
+		case "vlan":
+			circuit.Vlan = match
+		}
+	}
+	return circuit, nil
+}
+
+// FormatCircuitID is the CircuitID format we send in our Bootfile URL for ZTP devices
+func FormatCircuitID(circuit CircuitID) string {
```

Comment:
How about using a reference of the `CircuitID` to avoid copying the whole struct?

Reply:
I like it, I'll update, along with the above suggested comments.